### ISSUE TITLE
[9.0] Recurring document cannot be stopped nor deleted if Cron job is deleted

### ISSUE THREAD

**Issue by alaa9jo** [odoo/odoo#12207]
Impacted versions:
9.0
Steps to reproduce:
1)Create a new purchase order then create an invoice and pay it with cash
2)Create a new recurring type, type any name you want, and set Object to "Invoice"
3)Create a new recurring document, name it anything, set Source document to the recurring type you created earlier,then select the Bill number you created in first step
4)Save and your changes and notice how the STOP button appeared an a new cron job created automatically
5) click on edit, then click on the cursor that appears next to cron job name to open it's edit form
6) from the edit form,delete cron job,notice that the system returned back to edit form of recurring document but also notice that STOP button still there
7) Try to delete recurring document,it will show an error "You cannot delete an active subscription!" even though the cron job is deleted
8) Click on Stop, an error shows up:

```
Traceback (most recent call last):
  File "/odoo/odoo-server/openerp/http.py", line 643, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/odoo/odoo-server/openerp/http.py", line 680, in dispatch
    result = self._call_function(**self.params)
  File "/odoo/odoo-server/openerp/http.py", line 316, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/odoo/odoo-server/openerp/service/model.py", line 118, in wrapper
    return f(dbname, *args, **kwargs)
  File "/odoo/odoo-server/openerp/http.py", line 309, in checked_call
    result = self.endpoint(*a, **kw)
  File "/odoo/odoo-server/openerp/http.py", line 959, in __call__
    return self.method(*args, **kw)
  File "/odoo/odoo-server/openerp/http.py", line 509, in response_wrap
    response = f(*args, **kw)
  File "/odoo/odoo-server/addons/web/controllers/main.py", line 900, in call_button
    action = self._call_kw(model, method, args, {})
  File "/odoo/odoo-server/addons/web/controllers/main.py", line 888, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, *args, **kwargs)
  File "/odoo/odoo-server/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/odoo/odoo-server/addons/subscription/subscription.py", line 133, in set_done
    ids2 = [x['cron_id'][0] for x in res if x['id']]
TypeError: 'bool' object has no attribute '__getitem__'
```
Current behavior:
You cannot delete Recurring document nor stop it if cron job is deleted, this document is stuck forever

Expected behavior:
Either forbidden user from deleting cron job unless it's stopped first, or stop it automatically before deleting it,I should be able to go back and edit recurring document just fine and be able to create a new cron job

Video/Screenshot link (optional):

**Comment by Yenthe666** [comment]
@mart-e could you let somebody have a look at this one?

**Comment by mart-e** [comment]
Thanks for the report, indeed there was an issue, fixed at 440c5d6e443add49018b4615afe5a08613284bc3
